clients/admin-ui/src/features/data-discovery-and-detection/utils/getResourceName.ts

```diff
@@ -1,22 +1,47 @@
 import { DiscoveryMonitorItem } from "~/features/data-discovery-and-detection/types/DiscoveryMonitorItem";
 
-const TOP_LEVEL_FIELD_URN_PARTS = 5;
+const MAX_NON_NESTED_URN_LENGTH = 5;
+const URN_SEPARATOR = ".";
+/**
+ * Helper method for deriving a resource's display name from its URN
+ */
+const getResourceName = ({
+  name,
+  urn,
+  monitor_config_id,
+  database_name,
+  schema_name,
+  table_name,
+  top_level_field_name,
+}: DiscoveryMonitorItem) => {
+  const splitUrn = urn.split(URN_SEPARATOR);
 
-const getResourceName = (resource: DiscoveryMonitorItem) => {
-  const URN_SEPARATOR = ".";
-  const splitUrn = resource.urn.split(URN_SEPARATOR);
   if (
-    !resource.parent_table_urn ||
-    splitUrn.length === TOP_LEVEL_FIELD_URN_PARTS
+    !table_name ||
+    !top_level_field_name ||
```

Comment:
If it would be helpful I am adding in `top_level_field_urn` as well in this PR: https://github.com/ethyca/fidesplus/pull/1727/files#diff-8bd50e59212bea55bbd599dd3564999eecbda0c718c2681318bbc990a29d28c9R354

not sure if it's actually better or the same as `top_level_field_name` but I figured better to give as much info as possible now and then we can remove one later if it's not helpful

Reply:
I imagine `top_level_field_name` to be maybe a little tricky in structures like 
```
same_name
                  |------- same_name
                                                 |------- same_name
                                                                                 |------- same_name
```
etc etc

Reply:
Thanks for flagging-- yeah, that'll make this much cleaner if I can just elide that.

Reply:
`top_level_field_urn` is in now btw